extensions/quickfile/src/utils/hooks.ts

```diff
@@ -0,0 +1,46 @@
+import { showToast, Toast, LocalStorage } from "@raycast/api";
+import { useEffect, useState } from "react";
+
+export const useAsyncData = <T = any, I extends string | string[] | number[] = string>(
+  _key: I,
+  fetcher: (key: I) => Promise<T>
+) => {
+  const [data, setData] = useState<T>();
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<any>();
+
+  const key: string = Array.isArray(_key) ? _key.join(",") : _key;
+  useEffect(() => {
+    if (!key) return;
+
+    Promise.race([
+      LocalStorage.getItem(key).then((storedValue: unknown) => {
+        if (!storedValue || typeof storedValue !== "string") return;
+        return JSON.parse(storedValue) as T;
+      }),
+      fetcher(_key)
```

Comment:
We should only make one request per unique key ... Or are you thinking if the user invokes 'View Account Balances' several times in succession?

Reply:
> Or are you thinking if the user invokes 'View Account Balances' several times in succession?
that's one example, yeah 😸

But since this is a general util, it could be used in other places where multiple requests are plausible. Commonly this happens if `List.Item.Detail` dispatches an API call on mount; quickly moving selection up/down would end up sending a lot of API calls, despite the fact that once selection has changed, the result of the API call can be discarded.

But anyway, feel free to ignore my suggestion; we can always come back to it in the future if it somehow becomes necessary. 😉